backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/dsl/impl/ElasticsearchStringIndexFieldTypeOptionsStep.java

```diff
@@ -130,6 +129,7 @@ public ElasticsearchStringIndexFieldTypeOptionsStep aggregable(Aggregable aggreg
 		if ( analyzerName != null ) {
 			mapping.setType( DataTypes.TEXT );
 			mapping.setAnalyzer( analyzerName );
+			mapping.setSearchAnalyzer( searchAnalyzerName );
```

Comment:
You should probably add a check in the other branch of the `if`, to throw an exception if we have a search analyzer without an analyzer.
Same for the Lucene backend, I think.

Also, this would deserve a test in `org.hibernate.search.integrationtest.backend.tck.document.DocumentModelDslIT`, in my opinion.

Reply:
I think the same